buildserver/build_server.go

```diff
@@ -327,10 +349,27 @@ func (h *BuildHandler) Handle(ctx context.Context, conn *jsonrpc2.Conn, req *jso
 			}
 		}
 		rewriteURIFromClient := func(uri lsp.DocumentURI) lsp.DocumentURI {
-			if !strings.HasPrefix(string(uri), "file:///") {
-				return uri // refers to a resource outside of this workspace
+			var path string
+			if h.clientUsesFileSchemeWithinWorkspace {
+				if !strings.HasPrefix(string(uri), "file:///") {
+					return uri // refers to a resource outside of this workspace
+				}
+				path = strings.TrimPrefix(string(uri), "file://")
+			} else {
+				currentURL, err := url.Parse(string(uri))
+				if err != nil {
+					return uri
+				}
+				originalRootURI, err := url.Parse(string(h.init.OriginalRootURI))
+				if err != nil {
+					return uri
+				}
+				if currentURL.Hostname() == originalRootURI.Hostname() && currentURL.RawPath == originalRootURI.RawPath {
+					path = currentURL.Fragment
```

Comment:
shouldn't you be checking everything but fragment. IE so need to also include protocol. Maybe you could do something like

```go
path = currentURL.Fragment
currentURL.Fragment = ""
if *currentURL != *originalRootURI {
  return uri
}
```

Alternatively, why can't we just do `if strings.HasPrefix(uri, h.init.OriginalRootURI)`?

Reply:
✅ [Check everything but the fragment of the URL](https://github.com/sourcegraph/go-langserver/pull/373/commits/1c0778e6ba5a4db7af7d5b572218224828f7e42b)